packages/core/factories/src/useUserFactory.ts

```diff
@@ -0,0 +1,104 @@
+import { ref, Ref, computed } from '@vue/composition-api';
+import { UseUser } from '@vue-storefront/interfaces';
+
+export type UseUserFactoryParams<USER, UPDATE_USER_PARAMS> = {
+  loadUser: () => Promise<USER>;
+  logOut: (params?: {currentUser?: USER}) => Promise<void>;
+  updateUser: (params: {currentUser: USER; updatedUserData: UPDATE_USER_PARAMS}) => Promise<USER>;
+  register: (params: { email: string; password: string; firstName?: string; lastName?: string }) => Promise<USER>;
+  logIn: (params: { username: string; password: string }) => Promise<USER>;
+  changePassword: (params: {currentUser: USER; currentPassword: string; newPassword: string}) => Promise<USER>;
+};
+
+export function useUserFactory<USER, UPDATE_USER_PARAMS>(
```

Comment:
needed for registration
```suggestion
export function useUserFactory<USER, UPDATE_USER_PARAMS, REGISTER_USER_PARAMS >(
```

Reply:
Done